components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/util/converter/ObjectConverter.java

```diff
@@ -1147,15 +1151,18 @@ public static ApplicationBean convertStubApplicationContextToApplicationDefiniti
         if (applicationContext == null) {
             return null;
         }
-
+        Application application = ApplicationManager.getApplications().getApplication(applicationContext.getApplicationId());
         ApplicationBean applicationDefinition = new ApplicationBean();
         applicationDefinition.setApplicationId(applicationContext.getApplicationId());
         applicationDefinition.setAlias(applicationContext.getAlias());
         applicationDefinition.setMultiTenant(applicationContext.getMultiTenant());
         applicationDefinition.setName(applicationContext.getName());
         applicationDefinition.setDescription(applicationContext.getDescription());
-        applicationDefinition.setStatus(applicationContext.getStatus());
-
+        if (application.getStatus().name().equals(APPLICATION_STATUS_ACTIVE)) {
```

Comment:
Can't we reuse the ApplicationStatus enum without declaring the additional constant of APPLICATION_STATUS_ACTIVE?